packages/node/src/lib/metrics-log.ts

```diff
@@ -41,6 +41,37 @@ export interface LogResponse {
   ids: string | string[];
 }
 
+const BACKOFF_SECONDS = 15; // when we need to backoff HTTP requests, pause for seconds
+
+let backoffExpiresAt: Date;
+
+// Exported for use in unit tests
+export function setBackoff(expiresAt: Date | undefined) {
+  backoffExpiresAt = expiresAt;
+}
+
+function shouldBackoff(response: Response) {
+  // Some HTTP error codes indicate a problem with the API key, like the key is
+  // invalid or it's being rate limited. To avoid pointless requests to the
+  // ReadMe server, pause outgoing requests for a few seconds before trying
+  // again. Logs will be silently discarded while requests are paused, which is
+  // acceptable since the server wouldn't accept them anyway.
+  switch (response.status) {
+    case 401: // Unauthorized, i.e. this API key is invalid
+      return true;
+    case 403: // Forbidden, i.e. this API key is blocked by the server
+      return true;
```

Comment:
I feel like in the 401/403 case, we should actually surface the error (either by throwing or a `console.error`)... but I think that raises a few philosophical questions about the whole "fail silently" approach that we've been taking with these SDKs that warrants a broader discussion 🤔 for the sake of this PR, this LGTM!